### ISSUE TITLE
basename instead of absolute path

### ISSUE THREAD

**Issue by ovitus** [Shizcow/dmenu-rs#62]
Have this on NixOS, is there a configuration to show basenames instead of full paths?

**Comment by benjaminedwardwebb** [comment]
I would take a look at https://github.com/Shizcow/dmenu-rs/issues/53 and the linked PRs.

The tl;dr is that the full paths you're seeing are due to a bug that's since been fixed on `master`. Showing basenames is the expected behavior.

I haven't updated the package in nixpkgs (thanks for this reminder to do so, hopefully before 23.11 is cut), but since you're using NixOS you can use an overlay on the package so that it builds off `master` where the bug is fixed (that's what I've been using).

I'll point you here for information on nix overlays, in case you're unfamiliar: https://nixos.wiki/wiki/Overlays

Update: I submitted [this PR][1] to nixpkgs to upgrade the package to `5.5.3`.

[1]: https://github.com/NixOS/nixpkgs/pull/268547

**Comment by benjaminedwardwebb** [comment]
Here's an example NixOS configuration snippet that applies such an overlay.

```nix
nixpkgs.overlays = [
    (self: super: {
        dmenu-rs = super.dmenu-rs.overrideAttrs rec {
            version = "master";
            patches = [ ];
            src = super.pkgs.fetchFromGitHub {
                owner = "Shizcow";
                repo = super.pkgs.dmenu-rs.pname;
                rev = version;
                sha256 = "sha256-LdbTvuq1IbzWEoASscIh3j3VAHm+W3UekJNiMHTxSQI=";
            };
        };
    })
];
```

It's not precisely what I use, but I think it should work. Let me know if you try it and it gives you some trouble.

**Comment by ovitus** [comment]
Thanks for the quick response and an example overlay, I've been looking to try that out since I'm fairly new to Nix. I'll test it out tomorrow.

**Comment by ovitus** [comment]
Reporting back that the overlay worked. Thank you!

**Comment by benjaminedwardwebb** [comment]
> Reporting back that the overlay worked. Thank you!

Since this issue is a duplicate of a known, existing bug that is already fixed on `master`, can you close it @ovitus?

If you feel you'd like to record this issue somewhere, I would suggest filing an issue on [nixpkgs][2]. Or you can simply follow [this PR][1] until it merges (I'll try to shepherd the patch through in the next couple weeks when I have spare time). The patch will probably be released in NixOS 24.05 (and the overlay should work until then).

[1]: https://github.com/NixOS/nixpkgs/pull/268547
[2]: https://github.com/NixOS/nixpkgs

**Comment by benjaminedwardwebb** [comment]
@ovitus FYI an issue was filed to nixpkgs https://github.com/NixOS/nixpkgs/pull/223667
and the package update was merged https://github.com/NixOS/nixpkgs/pull/268547

It's not available in nixpkgs 23.11 but should be available on the next release 24.05 in May. Can you close this issue? Thanks! FYI @Shizcow

**Comment by ovitus** [comment]
Thank you @benjaminedwardwebb
